topography/include/gaussian_topog.inc

```diff
@@ -123,151 +88,63 @@ integer :: n
-!! Gaussian-shaped mountain.
-!!
-!> The height, position, width, and elongation of the mountain
-!! is controlled by optional arguments.
-!! @param real lon The mean grid box longitude in radians.
-!! @param real lat The mean grid box latitude in radians.
-!! @param real height Maximum surface height in meters.
-!! @param real olond, olatd Position/origin of mountain in degrees longitude and latitude.
-!! This is the location of the maximum height.
-!! @param real wlond, wlatd Gaussian half-width of mountain in degrees longitude and latitude.
-!! @param real rlond, rlatd Ridge half-width of mountain in degrees longitude and latitude.
-!! This is the elongation of the maximum height.
-!! @param real zsurf The surface height (in meters).
-!! The size of the returned field is size(lon) by size(lat).
-!!   </OUT>
-!!
-!! @throws FATAL shape(zsurf) is not equal to (/size(lon),size(lat)/)
-!!     Check the input grid size and output field size.
-!!     The input grid is defined at the midpoint of grid boxes.
-!!
-!! @note
-!!     Mountains do not wrap around the poles.
-!
-!! <br>Example usage:
-!! @code{.F90} zsurf = <B>get_gaussian_topog</B> ( lon, lat, height
-!!                    [, olond, olatd, wlond, wlatd, rlond, rlatd ] )@endcode
-function get_gaussian_topog ( lon, lat, height,                          &
-                              olond, olatd, wlond, wlatd, rlond, rlatd ) &
-                     result ( zsurf )
-
-real, intent(in)  :: lon(:), lat(:)
-real, intent(in)  :: height
-real, intent(in), optional :: olond, olatd, wlond, wlatd, rlond, rlatd
-real :: zsurf(size(lon,1),size(lat,1))
+real(kind=FMS_TOP_KIND_), intent(in)  :: lon(:), lat(:)
+real(kind=FMS_TOP_KIND_), intent(in)  :: height
+real(kind=FMS_TOP_KIND_), intent(in), optional :: olond, olatd, wlond, wlatd, rlond, rlatd
+real(kind=FMS_TOP_KIND_) :: zsurf(size(lon,1),size(lat,1))
 
 integer :: i, j
-real    :: olon, olat, wlon, wlat, rlon, rlat
-real    :: tpi, dtr, dx, dy, xx, yy
+real(kind=FMS_TOP_KIND_)    :: olon, olat, wlon, wlat, rlon, rlat
+real(kind=FMS_TOP_KIND_)    :: tpi, dtr, dx, dy, xx, yy
+integer, parameter     :: lkind = FMS_TOP_KIND_
 
   if (do_nml) call read_namelist
 
 ! no need to compute mountain if height=0
-  if ( height == 0. ) then
-       zsurf(:,:) = 0.
+  if ( height == 0.0_lkind) then
+       zsurf(:,:) = 0.0_lkind
        return
   endif
 
-  tpi = 2.0*pi
-  dtr = tpi/360.
+  tpi = 2.0_lkind*real(pi, FMS_TOP_KIND_)
+  dtr = tpi/360.0_lkind
 
 ! defaults and convert degrees to radians (dtr)
-  olon = 90.*dtr;  if (present(olond)) olon=olond*dtr
-  olat = 45.*dtr;  if (present(olatd)) olat=olatd*dtr
-  wlon = 15.*dtr;  if (present(wlond)) wlon=wlond*dtr
-  wlat = 15.*dtr;  if (present(wlatd)) wlat=wlatd*dtr
-  rlon =  0.    ;  if (present(rlond)) rlon=rlond*dtr
-  rlat =  0.    ;  if (present(rlatd)) rlat=rlatd*dtr
+  olon = 90.0_lkind*dtr;  if (present(olond)) olon=real(olond*dtr,r8_kind)
```

Comment:
sorry missed this one, olon= 90.0_r8_kind, and for the rest below too

Reply:
fixed in 7be92cd7